File: client-app/src/app/models/post.ts
import { Profile } from "./profile";

export interface Post {
  id: string;
  title: string;
  date: Date | null;
  description: string;
  category: string;
  city: string;
  venue: string;
  hostUsername: string;
  isCancelled: boolean;
  isGoing: boolean;
  isHost: boolean;
  host?: Profile;
  attendees: Profile[];
}

export class Post implements Post {
  constructor(init?: PostFormValues) {
    Object.assign(this, init);
  }
}

export class PostFormValues {
  id?: string = undefined;
  title: string = "";
  category: string = "";
  description: string = "";
  date: Date | null = null;
  city: string = "";
  venue: string = "";

  constructor(post?: PostFormValues) {
    if (post) {
      this.id = post.id;
      this.title = post.title;
      this.category = post.category;
      this.description = post.description;
      this.date = post.date;
      this.city = post.city;
      this.venue = post.venue;
    }
  }
}

File: Domain/Post.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public DateTime Date { get; set; }
        public string City { get; set; }
        public string Venue { get; set; } 
        public bool IsCancelled { get; set; }
        public ICollection<PostsAttendee> Attendees { get; set; } = new List<PostsAttendee>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}